library/alloc/src/boxed.rs

```diff
@@ -346,9 +346,13 @@ impl<T, A: Allocator> Box<T, A> {
     /// ```
     #[cfg(not(no_global_oom_handling))]
     #[unstable(feature = "allocator_api", issue = "32838")]
+    #[rustc_const_unstable(feature = "const_box", issue = "none")]
     #[must_use]
     #[inline]
-    pub fn new_in(x: T, alloc: A) -> Self {
+    pub const fn new_in(x: T, alloc: A) -> Self
+    where
+        A: ~const Allocator + ~const Drop,
```

Comment:
Is it possible to specify these bounds on the impl instead of each function?

Reply:
It's impossible, it seems that inherent impls with `~const` bounds is not allowed.
[Play](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=e2fd752957176b98dede2ab9514eb06e)